src/plugins/content_management/public/content_client/query_observable.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  notifyManager,
+  QueryObserver,
+  QueryObserverOptions,
+  QueryObserverResult,
+  QueryClient,
+} from '@tanstack/react-query';
+import { Observable } from 'rxjs';
+import { QueryKey } from '@tanstack/query-core/src/types';
+
+export const createQueryObservable = <
+  TQueryFnData = unknown,
+  TError = unknown,
+  TData = TQueryFnData,
+  TQueryData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey
+>(
+  queryClient: QueryClient,
+  queryOptions: QueryObserverOptions<TQueryFnData, TError, TData, TQueryData, TQueryKey>
+) => {
+  const queryObserver = new QueryObserver(
+    queryClient,
+    queryClient.defaultQueryOptions(queryOptions)
+  );
+
+  return new Observable<QueryObserverResult<TData, TError>>((subscriber) => {
+    const unsubscribe = queryObserver.subscribe(
+      // notifyManager is a singleton that batches updates across react query
```

Comment:
Strange I can't find anything about notifyManager in their docs... 🤔 

<img width="1257" alt="Screenshot 2023-01-30 at 13 52 49" src="https://user-images.githubusercontent.com/2854616/215496202-646be032-2443-4cae-990d-c32fbd34f735.png">

Reply:
I don't think it is a public api. It is exported from query-core and is used by react (or other libs) implementations. I learned about it from the useQuery hook source

Reply:
Ok. Isn't it risky to not use a public api?